File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length()!=t.length()) return false;

        int[] map = new int[26];

        for(int i=0; i<s.length(); i++){
            map[(int)s.charAt(i) - 'a']++;
            map[(int)t.charAt(i) - 'a']--;
        }

        for(int i:map){
            if(i!=0){
                return false;
            }
        }

        return true;

    }
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        freq_s = {}
        freq_t = {}

        for i in s:
            if i not in freq_s:
                freq_s[i] = 0
            freq_s[i] += 1

        for i in t:
            if i not in freq_t:
                freq_t[i] = 0
            freq_t[i] += 1

        if len(freq_s)!=len(freq_t):
            return False

        for i in s:
            if i not in freq_t:
                return False
            if freq_s[i] != freq_t[i]:
                return False

        return True